### PR TITLE
World config: Add modpack descriptions and remove dependencies there

### PR DESCRIPTION
**Pull request by HybridDog**
This allows adding a description.txt to modpacks which is shown in the world config menu.
Before the PR, dependencies of the modpack were shown, which is nonsense because modpacks don't have depends.

### CODE REVIEW

**Comment by SmallJoker** [comment]
How many modpacks do actually use `description.txt`? That file is only documented for mods, but not for modpacks.
Nevertheless, this information should be bundled together like in the (new) `mod.conf` file, which contains the dependencies and the description of the mod. It also allows adding more settings/values very easily.

**Comment by rubenwardy** [comment]
description.txt is currently supported and used for path, and quite a few do that.

modpack.txt should probably be renamed to modpack.conf and support for description and stuff added there. Maybe

File: builtin/mainmenu/dlg_config_world.lua [hunk 1]
```diff
@@ -33,14 +33,25 @@ local function get_formspec(data)
 
 	local hard_deps, soft_deps = pkgmgr.get_dependencies(mod.path)
 
+	if mod.is_modpack then
+		local info = pkgmgr.get_modpack_description(mod.path)
+		if info == "" then
+			info = fgettext("No modpack description provided.")
+		end
+		retval = retval ..
+			"label[0,0.7;" .. info .. "]"
```
Comment (rubenwardy) [inline builtin/mainmenu/dlg_config_world.lua]: Needs wrapping/scrollability - see the content tab for an implementation. Make sure to remove any backgrounds for it though

File: builtin/mainmenu/pkgmgr.lua [hunk 2]
```diff
@@ -299,6 +299,18 @@ function pkgmgr.render_packagelist(render_list)
 	return table.concat(retval, ",")
 end
 
+--------------------------------------------------------------------------------
+function pkgmgr.get_modpack_description(modfolder)
+	assert(modfolder, "missing modpack folder")
+	local descf = io.open(modfolder .. DIR_DELIM .. "description.txt", "r")
+	if not descf then
+		return "No description provided."
```
Comment (SmallJoker) [inline builtin/mainmenu/pkgmgr.lua]: Return `""`. Translation is handled above for the case when none is provided.

File: builtin/mainmenu/pkgmgr.lua [hunk 3]
```diff
@@ -299,6 +299,18 @@ function pkgmgr.render_packagelist(render_list)
...
+	end
+	local content = descf:read("*all")
+	descf:close()
+	return minetest.formspec_escape(content)
```
Comment (SmallJoker) [inline builtin/mainmenu/pkgmgr.lua]: Not implied by the function name. Escaping should be moved outside this function.

File: builtin/mainmenu/dlg_config_world.lua (hunk 1 above)
Comment (HybridDog) [reply builtin/mainmenu/dlg_config_world.lua]: fixed

**Comment by HybridDog** [comment]
I have addressed your comments.

File: builtin/mainmenu/dlg_config_world.lua [hunk 4]
```diff
@@ -33,14 +33,26 @@ local function get_formspec(data)
 
 	local hard_deps, soft_deps = pkgmgr.get_dependencies(mod.path)
 
+	if mod.is_modpack then
+		local info = minetest.formspec_escape(
+			pkgmgr.get_modpack_description(mod.path))
```
Comment (rubenwardy) [inline builtin/mainmenu/dlg_config_world.lua]: Please may you use   core.get_content_info(), and get the description from    info.description

Comment (HybridDog) [reply builtin/mainmenu/dlg_config_world.lua]: done

**Comment by Fixer-007** [comment]
>nonsense because modpacks don't have depends.

eh, ask @VanessaE on why her modpacks depend on that unified dies... oh well, too late

**Comment by VanessaE** [comment]
\<pedantic> None of my mod *packs* depend on it, though some of their components do.  \</pedantic> :stuck_out_tongue:

**Pull request merged**